src/bufrlib.h.in

```diff
@@ -102,42 +116,439 @@ typedef double f77r8; /* define the C object type that is equivalent to a REAL*8
 */
 void arallocc( void );
 void ardllocc( void );
-void bort( char *, size_t );
+
+/**
+ * Abort.
+ *
+ * @param c
+ * @param s
+ *
+ * @author J.Ator @date 2003-11-04
+ */
+void bort( char *c, size_t s);
 void bort_exit( void );
-void bvers( char *, size_t ); 
-void cadn30( f77int *, char *, size_t ); 
-void closbf( f77int * );
-int cmpia( const void *, const void * );
-void cmpmsg( char *, size_t );
-void elemdx( char *, f77int *, size_t );
-void gets1loc( char *, f77int *, f77int *, f77int *, f77int *, size_t );
-f77int ibfms( f77r8 * );
-f77int ichkstr ( char *, char *, f77int *, size_t, size_t );
-f77int icvidx ( f77int *, f77int *, f77int * );
-f77int ifxy( char *, size_t );
-f77int igetntbi( f77int *, char *, size_t );
-f77int igetprm( char *, size_t );
-f77int igettdi( f77int * );
-f77int imrkopr( char *, size_t );
-void ipkm( char *, f77int *, f77int *, size_t );
-f77int ireadns( f77int *, char *, f77int *, size_t );
-f77int istdesc( f77int * );
-f77int iupb( f77int *, f77int *, f77int * );
-f77int iupbs01 ( f77int *, char *, size_t );
-f77int iupm ( char *, f77int *, size_t );
-void maxout( f77int * );
-void nemtab( f77int *, char *, f77int *, char *, f77int *, size_t, size_t );
-void nemtbb( f77int *, f77int *, char *, f77int *, f77int *, f77int *, size_t );
-void nummtb( f77int *, char *, f77int * );
-void numtbd( f77int *, f77int *, char *, char *, f77int *, size_t, size_t );
-void openmb( f77int *, char *, f77int *, size_t );
-void pktdd( f77int *, f77int *, f77int *, f77int * );
-void restd( f77int *, f77int *, f77int *, f77int * );
-void stntbi( f77int *, f77int *, char *, char *, char *, size_t, size_t, size_t );
-void strnum( char *, f77int *, size_t );
-void stseq( f77int *, f77int *, f77int *, char *, char *, f77int *, f77int * );
-void ufbint( f77int *, f77r8 *, f77int *, f77int *, f77int *, char *, size_t );
-void ufbseq( f77int *, f77r8 *, f77int *, f77int *, f77int *, char *, size_t );
-void uptdd( f77int *, f77int *, f77int *, f77int * );
-void wrdesc( f77int, f77int *, f77int * );
+
+/**
```

Comment:
@jbathegit what are all these C prototype functions?

Reply:
These are prototypes to the library subprograms of the same name, some of which are C subprograms, but most of which are Fortran subprograms called from C.   The latter will go away once #79 is completed, b/c those will instead be handled via new BIND(C, ) attributes.  And all of the f77 typedefs will go away as well at that point.  But for now, since we're still using the old UNDERSCORE append way of linking C to Fortran, then that's why we have to list all of these here, along with supplemental size_t arguments for prior character string arguments.

For documentations purposes, is it enough to just say it's a prototype to the subprogram of the same name, since the subprogram documentation explains everything in further detail?  And for the variable names, could we just copy the same names and same definitions that are listed in the subprogram documentation?  Again, most of these prototypes will go away once I get a chance to work on #79.

Reply:
How about I just go ahead and approve and merge this PR as is, and I promise I'll fill in all of the bufrlib.h.in prototype information myself as I work on #79?  Again, there will be far fewer prototypes to deal with once that's done.

That way, it will free you and Alex up to move forward on documenting other areas of the library.  I'll agree to follow the formatting and examples you've laid out in this PR and get the rest of the C code up to snuff as I work on #79, if you guys can focus your efforts on doxygen updates for all of the remaining Fortran subprograms.  I'll remain available to answer questions if you have them, but this way (i.e. if you guys focus on the Fortran parts and I focus on the C parts), then we lessen the chances of stepping on each others' toes as we work in parallel.  All of the Fortran subprograms contain documentation, but the documentation just needs to be converted to doxygen-style, as you've seen.

Sound good?